gltfTutorial/gltfTutorial_006_MaterialsTechniquesProgramsShaders.md

```diff
@@ -0,0 +1,288 @@
+
+### Shaders
+
+In the context of glTF, the vertex- and fragment shader are usually contained in dedicated files, which are referred to by the [`shader`](https://github.com/KhronosGroup/glTF/tree/master/specification#reference-shader) objects using a URI:
+
+```javascript
+"shaders": {
+    "exampleVertexShader": {
+        "uri": "vertexShader.glsl",
+        "type": 35633
+    },
+    "exampleFragmentShader": {
+        "uri": "fragmentShader.glsl",
+        "type": 35632
+    }
+}
+```
+
+The shader source code that is referred to by the URI is stored as a plain text, as described in the section about [GLSL shader data in `shaders`](gltfTutorial_002_BasicGltfStructure.md#glsl-shader-data-in-shaders). This source code can be read into a string, and directly be passed to OpenGL, as shown in the implementation notes below. In addition to the URI, the `shader` object also contains the `type` of the shader, which simply is the respective OpenGL constant: Either `35633` (`GL_VERTEX_SHADER`) for vertex shaders, or `35632` (`GL_FRAGMENT_SHADER`) for fragment shaders.
+
+
+#### Attributes and uniforms in shaders
+
+The vertex shader may define different *attributes*. These attributes describe all properties of a single vertex. Recall this image, already shown in the section about [Mesh primitive attributes](gltfTutorial_005_MeshesTexturesImagesSamplers.md#mesh-primitive-attributes):
+
+![meshPrimitive accessors](images/meshPrimitiveAttributes.png)
+
+One can imagine that the vertex shader is executed for each vertex, and receives the attribute values for the respective vertex in the attribute variables. For example, the following vertex shader defines attributes for the position, normal and texture coordinate of each vertex:
+
+```glsl
+attribute vec3 a_position;
+attribute vec3 a_normal;
+attribute vec2 a_texcoord0;
+
+...
+uniform vec3 u_lightDirection;
+uniform mat4 u_modelViewMatrix;
+...
+
+void main(void)
+{
+    vec4 pos = u_modelViewMatrix * vec4(a_position,1);
+    ...
+    gl_Position = ...;
+}
+```
+
+Additionally, the vertex- and fragment shaders can define *uniforms*. These are further input parameters for the rendering process which are independent of the current vertex. For example, the vertex shader above defines uniforms for the light direction and the model-view-matrix. The following fragment shader defines additional uniforms for the ambient light color, and for a texture that will be used as the diffuse color of the material:
+
+```glsl
+...
+uniform vec4 u_ambientLightColor;
+uniform sampler2D u_diffuseTexture;
+...
+
+void main(void)

```

Comment:
Since there's no lighting, I think the use of `diffuse` here is misleading.  Perhaps rename it to something like decal or add a call to a lighting function (even if the implementation is not shown).